src/vmm/src/config/mod.rs

```diff
@@ -258,6 +264,32 @@ impl TryFrom<&str> for BlockConfig {
     }
 }
 
+/// Console configuration
+#[derive(Clone, Debug, PartialEq)]
+pub struct ConsoleConfig {
+    /// Type of console.
+    pub console_type: String,
+}
+
+impl TryFrom<&str> for ConsoleConfig {
+    type Error = ConversionError;
+
+    fn try_from(console_cfg_str: &str) -> Result<Self, Self::Error> {
+        // Supported options: `type=String`
+        let mut arg_parser = CfgArgParser::new(console_cfg_str);
+
+        let console_type = arg_parser
```

Comment:
I think we can also define an enum, let's say `ConsoleType`, for the possible values for `type` and if the string is not one of the expected values, throw an error. So ConsoleConfig will no longer store a `String`, but a `ConsoleType` object.

Reply:
Then we should also add tests for invalid input, example `--console type=blah`.